opslevel/resource_opslevel_check_code_issue.go

```diff
@@ -0,0 +1,323 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckCodeIssueResource{}
+	_ resource.ResourceWithImportState = &CheckCodeIssueResource{}
+)
+
+func NewCheckCodeIssueResource() resource.Resource {
+	return &CheckCodeIssueResource{}
+}
+
+// CheckCodeIssueResource defines the resource implementation.
+type CheckCodeIssueResource struct {
+	CommonResourceClient
+}
+
+var resolutionTimeType = map[string]attr.Type{
+	"unit":  types.StringType,
+	"value": types.Int64Type,
+}
+
+type CheckCodeIssueResourceModel struct {
+	CheckCodeBaseResourceModel
+
+	Constraint     types.String `tfsdk:"constraint"`
+	IssueName      types.String `tfsdk:"issue_name"`
+	IssueType      types.List   `tfsdk:"issue_type"`
+	MaxAllowed     types.Int64  `tfsdk:"max_allowed"`
+	ResolutionTime types.Object `tfsdk:"resolution_time"`
+	Severity       types.List   `tfsdk:"severity"`
+}
+
+func NewCheckCodeIssueResourceModel(ctx context.Context, check opslevel.Check, planModel CheckCodeIssueResourceModel) CheckCodeIssueResourceModel {
```

Comment:
Same as above - `givenModel` because sometimes it stateModel and sometimes its planModel